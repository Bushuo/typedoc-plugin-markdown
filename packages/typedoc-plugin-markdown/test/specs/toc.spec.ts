import * as Handlebars from 'handlebars';
import { Reflection } from 'typedoc';

import { TestApp } from '../test-app';

describe(`TOC:`, () => {
  let testApp: TestApp;
  let moduleReflection: Reflection;
  let classReflection: Reflection;

  beforeAll(() => {
    testApp = new TestApp(['breadcrumbs.ts']);
  });

  describe(`(default)`, () => {
    beforeAll(async () => {
      await testApp.bootstrap();
      moduleReflection = testApp.project.children[0];
      classReflection = testApp.project.findReflectionByName('Breadcrumbs');
    });

    test(`should display toc for module'`, () => {
      expect(
        TestApp.compileHelper(Handlebars.helpers.toc, moduleReflection),
      ).toMatchSnapshot();
    });

    test(`should display toc for class'`, () => {
      expect(
        TestApp.compileHelper(Handlebars.helpers.toc, classReflection),
      ).toMatchSnapshot();
    });
  });
  describe(`(hideInPageToc)`, () => {
    beforeAll(async () => {
      await testApp.bootstrap({ hideInPageTOC: true });
      moduleReflection = testApp.project.children[0];
      classReflection = testApp.project.findReflectionByName('Breadcrumbs');
    });

    test(`should not display toc for class'`, () => {
      expect(Handlebars.helpers.toc.call(classReflection)).toBeNull();
    });
  });
});
