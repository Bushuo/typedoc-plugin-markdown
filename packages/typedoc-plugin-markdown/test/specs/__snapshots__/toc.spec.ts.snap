// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOC: (default) should display toc for class' 1`] = `
"## Table of contents

### Constructors

- [constructor](breadcrumbs.md#constructor)

### Methods

- [someMethod](breadcrumbs.md#somemethod)
"
`;

exports[`TOC: (default) should display toc for module' 1`] = `
"## Table of contents

### Constructors

- [constructor](breadcrumbs.md#constructor)

### Methods

- [someMethod](breadcrumbs.md#somemethod)
"
`;
